### PR TITLE
add filter to validate CITATION.cff using Docker image

### PR DESCRIPTION
**Pull request by fdiblen**
**Description**

**Related issues**:
- #4 

**Instructions to review the pull request**

- Make a GitHub Access token here https://github.com/settings/tokens
  - Make sure you copy it when it appears on screen, because you can't see it again. You can reset it, though.
- export GITHUB_TOKEN as an environment variable with the token above
- install Node >= 14

    ```
    cd $(mktemp -d --tmpdir cffbot-XXXXXX)
    git clone https://github.com/cffbots/filtering .
    git checkout <this-branch>
    npm install
    node query.mjs
    ```

**Sample filtered list**

urls:  [
  'https://github.com/e-mental-health/data-processing',
  'https://github.com/NLeSC/python-template',
  'https://github.com/NLeSC/case-law-app',
  'https://github.com/NLeSC/eEcology-Annotation-WS',
  'https://github.com/NLeSC/eEcology-script-wrapper',
  'https://github.com/NLeSC/eEcology-SMS-reciever',
  'https://github.com/NLeSC/esibayes'
]

### CODE REVIEW

File: query.mjs [hunk 1]
```diff
@@ -105,11 +144,13 @@ const inactivity_threshold = 12 * 30 * 24 * 60 * 60 * 1000 // X months in millis
 const octokit = new Octokit({auth: process.env.GITHUB_TOKEN});
 
 let urls = urls_rsd;
+urls = urls.filter(includeWhitelisted);
```
Comment (abelsiqueira) [inline query.mjs:147]: The `whitelist` variable was removed, so this is failing.

**Review by jspaaks — APPROVED** [review]
Thanks for the PR

File: query.mjs [hunk 2]
```diff
@@ -26,6 +26,45 @@ const includeHasCitationcff = async (url) => {
...
+        result = await execPromise(dockerCommand);
+        // console.log('\n\n===============================')
+        // console.log('Success: ', url, default_branch);
+        if (result.includes(checkValidString)) {
+            return true;
+        }
+    } catch (error) {
+        // console.log('\n\n===============================')        
+        // console.error('Failed: ', url, default_branch); 
+        // console.error('Error: ', error);
```
Comment (fdiblen) [inline query.mjs:62]:
```suggestion
```

File: query.mjs (hunk 1 above)
Comment (abelsiqueira) [inline query.mjs:147]:
```suggestion
```

File: query.mjs [hunk 3]
```diff
@@ -26,6 +26,45 @@ const includeHasCitationcff = async (url) => {
...
+                resolve(stdout.trim());
+            });
+        });
+    }
+
+    let result;
+    try {
+        result = await execPromise(dockerCommand);
+        // console.log('\n\n===============================')
+        // console.log('Success: ', url, default_branch);
```
Comment (abelsiqueira) [inline query.mjs:55]:
```suggestion
```

File: query.mjs (hunk 2 above)
Comment (abelsiqueira) [inline query.mjs:62]:
```suggestion
```

**Review by abelsiqueira — CHANGES_REQUESTED** [review]

**Review by abelsiqueira — APPROVED** [review]
Thanks for the PR. LGTM. Tested locally and it returned 17 repos.

**Comment by fdiblen** [comment]
Thank you @jspaaks and @abelsiqueira for reviewing!

**Pull request merged**